File: java-api/src/main/java/com/fagerland/javaapi/config/JwtConfig.java
package com.fagerland.javaapi.config;

import com.fagerland.javaapi.filters.JwtFilter;
import org.springframework.security.config.annotation.SecurityConfigurerAdapter;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.web.DefaultSecurityFilterChain;
import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;

public class JwtConfig extends SecurityConfigurerAdapter<DefaultSecurityFilterChain, HttpSecurity> {

    public JwtConfig() {

    }

    @Override
    public void configure(HttpSecurity httpSecurity) throws Exception {
        JwtFilter filter = new JwtFilter();
        httpSecurity.addFilterBefore(filter, UsernamePasswordAuthenticationFilter.class);
    }
}

File: kotlin-maven-api/src/main/kotlin/com/fagerland/brotime/config/JwtConfig.kt
package com.fagerland.brotime.config

import com.fagerland.brotime.filters.JwtFilter
import com.fagerland.brotime.services.JwtService
import org.springframework.security.config.annotation.SecurityConfigurerAdapter
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.web.DefaultSecurityFilterChain
import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter

class JwtConfig(
    private val jwtService: JwtService
) : SecurityConfigurerAdapter<DefaultSecurityFilterChain, HttpSecurity>() {

    override fun configure(httpSecurity: HttpSecurity) {
        val filter = JwtFilter(jwtService)
        httpSecurity.addFilterBefore(filter, UsernamePasswordAuthenticationFilter::class.java)
    }
}
